### DISCUSSION TITLE
Publish On IPFS using FVM smart Contract

### DISCUSSION THREAD

**Discussion by dipanshuhappy** [filecoin-project/community#604] [Developers]
Hey , i am very new to this FVM and i wanted to know what it can do and what it can't do . If you can't upload files through FVM then how can you do it without FVM . Also how do you make a ownership and access control through FVM .

**Comment by ZakAyesh** [comment]
Hi! The FVM allow users to deploy smart contracts to the Filecoin blockchain. Currently the FEVM (the Ethereum Virtual Machine on Filecoin) allows you to do this by writing smart contracts in a language that compiles to EVM bytecode, such as the popular language Solidity. The FEVM is actually built on top of the FVM, and is currently the only runtime that is supported! Also please note that IPFS and Filecoin are separate networks (although they share a lot of the same technology)

For now files uploaded are done outside the FVM as described here:
https://github.com/filecoin-project/community/discussions/601

You can also checkout the Spacewarp cheatsheet for a good overview of the ecosystem and some possible design patterns:
https://github.com/filecoin-project/community/discussions/585

Hope that helps!

**Reply by dipanshuhappy** [reply]
Yes it helps ...so basically Filecoin is like a big market place for storage provider and clients .So what are the core functionalities of the FVM?
